usermods/Internal_Temperature_v2/usermod_internal_temperature.h

```diff
@@ -6,21 +6,34 @@ class InternalTemperatureUsermod : public Usermod
 {
 
 private:
+  static const unsigned long minLoopInterval = 1000;  // minimum allowable interval (ms)
   unsigned long loopInterval = 10000;
   unsigned long lastTime = 0;
   bool isEnabled = false;
   float temperature = 0;
+  int presetToActivate = -1;            // Preset to activate when temp goes above threshold (-1 = disabled)
+  float activationThreshold = 95.0;     // Temperature threshold to trigger high-temperature actions
+  float resetMargin = 2.0;              // Margin below the activation threshold (Prevents frequent toggling when close to threshold)
+  bool isAboveThreshold = false;        // Flag to track if the high temperature preset is currently active
 
   static const char _name[];
   static const char _enabled[];
   static const char _loopInterval[];
+  static const char _activationThreshold[];
+  static const char _presetToActivate[];
 
   // any private methods should go here (non-inline method should be defined out of class)
   void publishMqtt(const char *state, bool retain = false); // example for publishing MQTT message
 
+  //  Makes sure the measurement interval can't be set too low
+  void setSafeLoopInterval(unsigned long newInterval) {
```

Comment:
There is no benefit to this as it is inlined by compiler.

Reply:
I saw this done in the Battery usermod, would you like me to remove it?

Reply:
Usermod's `loop()` is executed as many times as possible with no predetermined frequency. It is up to you to "schedule" *regular* processing within it.

With this comment I was referring to the fact that any function definition in the definition of a class is inlined as if you would add `inline` keyword. To avoid this you can define the function outside class definition or, in this case, simply use assignment in places where function is called (this not being wise always).

Reply:
Thankyou for the helpfull explination Blaz... Much appreciated! :)